### PR TITLE
apimachinery: remove pre-apigroups import path logic

### PR DESCRIPTION
**Pull request by sttts**

### CODE REVIEW

File: pkg/api/meta/install.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
+package meta
```
Comment (deads2k) [inline pkg/api/meta/install.go]: `pkg/api/legacy`?

Comment (sttts) [inline pkg/api/meta/install.go]: Sure, didn't know where to put it.

File: pkg/api/meta/install.go [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+	registry.RegisterVersions([]schema.GroupVersion{coreV1})
+	if err := registry.EnableVersions(coreV1); err != nil {
+		glog.Fatal(err)
+	}
+
+	// register types as core v1
+	if err := addToCore(scheme); err != nil {
+		glog.Fatal(err)
+	}
+	if err := addToCoreV1(kapi.Scheme); err != nil {
```
Comment (deads2k) [inline pkg/api/meta/install.go]: Why is this scheme hardcoded?

File: pkg/api/meta/install.go [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+var (
+	accessor = meta.NewAccessor()
+	coreV1   = schema.GroupVersion{Group: "", Version: "v1"}
+)
+
+func InstallLegacy(group string, addToCore, addToCoreV1 func(*runtime.Scheme) error, groupFactoryRegistry announced.APIGroupFactoryRegistry, registry *registered.APIRegistrationManager, scheme *runtime.Scheme) {
+	groupMetaFactory, found := groupFactoryRegistry[group]
+	if !found {
+		glog.Fatalf("Group %q not found", group)
```
Comment (deads2k) [inline pkg/api/meta/install.go]: we normally panic, right?

Comment (sttts) [inline pkg/api/meta/install.go]: do we? will change that.

File: pkg/api/meta/install.go (hunk 2 above)
Comment (sttts) [inline pkg/api/meta/install.go]: it shouldn't

**Comment by deads2k** [comment]
Minor comments.  After they are fixed up, this lgtm

File: pkg/api/meta/install.go (hunk 2 above)
Comment (sttts) [inline pkg/api/meta/install.go]: fixed

File: pkg/api/meta/install.go (hunk 3 above)
Comment (sttts) [inline pkg/api/meta/install.go]: fixed

**Comment by deads2k** [comment]
@sttts I happened to be near here again today. this is on my wishlist.

**Comment by enj** [comment]
The client seems to be defaulting to wrong group:

```
=== BEGIN TEST CASE ===
test/cmd/basicresources.sh:313: executing 'oc run --dry-run foo --image=bar -o "go-template={{.kind}} {{.apiVersion}}"' expecting success and text 'DeploymentConfig v1'
FAILURE after 0.258s: test/cmd/basicresources.sh:313: executing 'oc run --dry-run foo --image=bar -o "go-template={{.kind}} {{.apiVersion}}"' expecting success and text 'DeploymentConfig v1': the output content test failed
Standard output from the command:
DeploymentConfig apps.openshift.io/v1
There was no error output from the command.
=== END TEST CASE ===
```

`DeploymentConfig apps.openshift.io/v1`


```
=== BEGIN TEST CASE ===
test/cmd/newapp.sh:85: executing 'oc new-app nginx -l qwer1345%$$#=self' expecting failure and text 'error: ImageStream "nginx" is invalid'
FAILURE after 0.421s: test/cmd/newapp.sh:85: executing 'oc new-app nginx -l qwer1345%$$#=self' expecting failure and text 'error: ImageStream "nginx" is invalid': the output content test failed
Standard output from the command:
--> Found Docker image b8efb18 (3 weeks old) from Docker Hub for "nginx"

    * An image stream will be created as "nginx:latest" that will track this image
    * This image will be deployed in deployment config "nginx"
    * Port 80/tcp will be load balanced by service "nginx"
      * Other containers can access this service through the hostname "nginx"
    * WARNING: Image "nginx" runs as the 'root' user which may not be permitted by your cluster administrator

--> Creating resources with label  ...
--> Failed

Standard error from the command:
    error: ImageStream.image.openshift.io "nginx" is invalid: metadata.labels: Invalid value: "qwer1345%9070#": name part must consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character (e.g. 'MyName',  or 'my.name',  or '123-abc', regex used for validation is '([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9]')
    error: DeploymentConfig.apps.openshift.io "nginx" is invalid: [metadata.labels: Invalid value: "qwer1345%9070#": name part must consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character (e.g. 'MyName',  or 'my.name',  or '123-abc', regex used for validation is '([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9]'), spec.template.labels: Invalid value: "qwer1345%9070#": name part must consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character (e.g. 'MyName',  or 'my.name',  or '123-abc', regex used for validation is '([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9]')]
    error: Service "nginx" is invalid: [metadata.labels: Invalid value: "qwer1345%9070#": name part must consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character (e.g. 'MyName',  or 'my.name',  or '123-abc', regex used for validation is '([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9]'), spec.selector: Invalid value: "qwer1345%9070#": name part must consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character (e.g. 'MyName',  or 'my.name',  or '123-abc', regex used for validation is '([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9]')]
=== END TEST CASE ===
```

`ImageStream.image.openshift.io "nginx" is invalid`

**Comment by sttts** [comment]
@enj thanks for investigating. Have to recheck whether I missed some special case in the legacy code I dropped and replaced by the generic registration.

**Comment by deads2k** [comment]
@sttts consistently failing test-cmd

**Comment by deads2k** [comment]
I opened  #15955 to fix the compile error and it hit a serialization change in `oc run` and failed test-cmd.

**Pull request closed without merging**